### ISSUE TITLE
Zipping two streams interferes with bracket

### ISSUE THREAD

**Issue by sirocchj** [functional-streams-for-scala/fs2#1130]
It appears that if the release handle completes with error this will not bubble up unless it's coming from the "left" stream.

```scala
import cats.effect.IO, fs2.Stream

val s1 = Stream.bracket(IO(1))(Stream.emit(_), _ => IO.pure(()))
val s2 = Stream.bracket(IO("a"))(Stream.emit(_), _ => IO.raiseError(new RuntimeException("boom")))

s1.compile.drain.unsafeRunSync // ok!
s2.compile.drain.unsafeRunSync // ok! java.lang.RuntimeException: boom

s1.zip(s2).compile.drain.unsafeRunSync // nothing... odd to me
s2.zip(s1).compile.drain.unsafeRunSync // ok! java.lang.RuntimeException: boom
```

**Comment by pchlupacek** [comment]
does that behave same if you do not `drain` the results?

**Comment by sirocchj** [comment]
Nope, it blows (tried with `.last`)

**Comment by sirocchj** [comment]
Yes, meaning that `s1.zip(s2).compile.last.unsafeRunSync` returns fine (`Some((1,a))`) without blowing up due to `s2`'s release throwing

**Comment by sirocchj** [comment]
OTOH, `s2.zip(s1).compile.last.unsafeRunSync` blows up as when `.drain`ing

**Comment by AdamChlupacek** [comment]
@sirocchj yeah its just an issue of closing the root scope, we ignore any throwable coming from there. This is gonna be simple fix.

**Comment by mpilquist** [comment]
Fixed in #1131
